### PR TITLE
inference-triton-cli-experimental

### PR DESCRIPTION
**Pull request by ArzooAneja**
# PR into Azure/azureml-examples

## Checklist

I have:

- [ ] read and followed the contributing guidelines

## Changes

-

fixes #

### CODE REVIEW

**Review by msjelly — CHANGES_REQUESTED** [review]
Notebook checkpoint files (.ipynb_checkpoints) should be removed from the Pull Request.

File: cli/endpoints/online/triton/single-model/single-triton-lite.ipynb [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+      "Requirement already satisfied: requests-oauthlib>=0.5.0 in /home/arzooaneja/anaconda3/envs/azmlenv/lib/python3.7/site-packages (from msrest>=0.6.18->azure-storage-blob) (1.3.0)\n",
+      "Requirement already satisfied: idna<3,>=2.5 in /home/arzooaneja/anaconda3/envs/azmlenv/lib/python3.7/site-packages (from requests>=2.18.4->azure-core<2.0.0,>=1.10.0->azure-storage-blob) (2.10)\n",
+      "Requirement already satisfied: chardet<5,>=3.0.2 in /home/arzooaneja/anaconda3/envs/azmlenv/lib/python3.7/site-packages (from requests>=2.18.4->azure-core<2.0.0,>=1.10.0->azure-storage-blob) (4.0.0)\n",
+      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in /home/arzooaneja/anaconda3/envs/azmlenv/lib/python3.7/site-packages (from requests>=2.18.4->azure-core<2.0.0,>=1.10.0->azure-storage-blob) (1.26.4)\n",
+      "Requirement already satisfied: oauthlib>=3.0.0 in /home/arzooaneja/anaconda3/envs/azmlenv/lib/python3.7/site-packages (from requests-oauthlib>=0.5.0->msrest>=0.6.18->azure-storage-blob) (3.1.0)\n"
```
Comment (msjelly) [inline cli/endpoints/online/triton/single-model/single-triton-lite.ipynb:35]: Include output only where it's helpful to the user.  Verbose output like this should be removed / cleaned up throughout the notebook.

File: cli/endpoints/online/triton/single-model/single-triton-lite.ipynb [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "subscripton = \"13e50845-67bc-4ac5-94db-48d493a6d9e8\"\n",
+    "resource_group = \"chrjia-rg\"\n",
+    "workspace = \"chrjia-eastus2euap2\"\n",
```
Comment (msjelly) [inline cli/endpoints/online/triton/single-model/single-triton-lite.ipynb:85]: Remove actual subscription GUIDs, resource group names.  This is a general comment for all recipes.

File: cli/endpoints/online/triton/single-model/single-triton-lite.ipynb [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+      "  \"id\": \"azureml:/subscriptions/13e50845-67bc-4ac5-94db-48d493a6d9e8/resourceGroups/chrjia-rg/providers/Microsoft.MachineLearningServices/workspaces/chrjia-eastus2euap2/models/densenet-onnx/versions/1\",\n",
+      "  \"name\": \"densenet-onnx\",\n",
+      "  \"path\": \"az-ml-artifacts/57396fa60078750da5128ce974a4a65f/models\",\n",
+      "  \"properties\": {},\n",
+      "  \"resourceGroup\": \"chrjia-rg\",\n",
+      "  \"tags\": {},\n",
+      "  \"version\": 1\n",
+      "}\n",
+      "\u001b[0m"
+     ]
```
Comment (msjelly) [inline cli/endpoints/online/triton/single-model/single-triton-lite.ipynb:121]: See comments above.  Clean up output from published recipe.

**Comment by msjelly** [comment]
In this example  from managed online endpoint recipe ([link](https://github.com/Azure/azureml-examples/blob/cli-preview/cli/endpoints/online/managed/simple-flow/1-create-endpoint-with-blue.ym)), the deployment filename is descriptive.

Our deployment.yml can have a more descriptive names.  
create-endpoint-on-aks.yml  (renamed from deployment.yml)
create-endpoint-on-mir-gpu.yml (renamed from mir_gpu_deployment.yml)

File: cli/endpoints/online/triton/single-model/mir_gpu_deployment.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+name: mirsingle4
```
Comment (msjelly) [inline cli/endpoints/online/triton/single-model/mir_gpu_deployment.yml:1]: 'single-mir' for consistent format with 'single-aks' above.

File: cli/endpoints/online/triton/single-model/mir_gpu_deployment.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+        model: azureml:densenet-onnx:1
+        code_configuration:
+            code:
+                local_path: ./code
+            scoring_script: dummy.py
+        environment:
+            name: single-env
+            version: 1
+            docker:
+                image: metricstestacr.azurecr.io/triton-base:3
```
Comment (msjelly) [inline cli/endpoints/online/triton/single-model/mir_gpu_deployment.yml:17]: FYI. image name string will be updated later once triton image is published to MCR.

File: cli/endpoints/online/triton/single-model/single-triton-lite.ipynb [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "subscripton = \"13e50845-67bc-4ac5-94db-48d493a6d9e8\"\n",
+    "resource_group = \"chrjia-rg\"\n",
+    "workspace = \"chrjia-eastus2euap2\"\n",
+    "model_name = \"densenet-onnx\"\n",
+    "service_name = \"lite14\""
```
Comment (msjelly) [inline cli/endpoints/online/triton/single-model/single-triton-lite.ipynb:87]: For DevPlatV2 notebooks, replace 'service_name' wth 'endpoint_name' and '$service_name' with '$endpoint_name'.

**Comment by msjelly** [comment]
In both notebooks (ipynb), remove output from !pip install azure-storage-blob.

**Comment by msjelly** [comment]
Rename files and make corresponding changes inside the notebooks (ipynb) where the filenames are referenced.
deployment.yml => endpoint-aks.yml
mir_gpu_deployment.yml => endpoint-mir-gpu.yml

**Comment by msjelly** [comment]
Rename notebook files.
single-triton-lite-AKS.ipynb => single-model-aks.ipynb
single-triton-lite-AKS.ipynb => single-model-managed.ipynb

**Comment by msjelly** [comment]
In endpoint-aks.yml (currently deployment.yml), change target to azureml:aks-test.  
In notebook single-model-aks.ipynb,  change comment following "Create endpoint":
Deploy to a pre created AKS compute aks-test.

File: cli/endpoints/online/triton/single-model/single-triton-lite.ipynb (hunk 2 above)
Comment (ArzooAneja) [reply cli/endpoints/online/triton/single-model/single-triton-lite.ipynb]: Issue Resolved

File: cli/endpoints/online/triton/single-model/single-triton-lite.ipynb (hunk 1 above)
Comment (ArzooAneja) [reply cli/endpoints/online/triton/single-model/single-triton-lite.ipynb]: Done

File: cli/endpoints/online/triton/single-model/single-triton-lite.ipynb (hunk 3 above)
Comment (ArzooAneja) [reply cli/endpoints/online/triton/single-model/single-triton-lite.ipynb]: Done

File: cli/endpoints/online/triton/single-model/single-triton-lite.ipynb (hunk 6 above)
Comment (ArzooAneja) [reply cli/endpoints/online/triton/single-model/single-triton-lite.ipynb]: Done

**Comment by msjelly** [comment]
In all .ipynb notebooks, replace string 'service_name' wth 'endpoint_name' and '$service_name' with '$endpoint_name'.

**Comment by msjelly** [comment]
Please drop 'experimental' from branch name and rename branch:  triton-inference-cli.

**Comment by msjelly** [comment]
Please remove file single-model/src/__pycache__/model_utils.cpython-37.pyc.
Please do not include filename with .pyc extension.

**Pull request closed without merging**